contractcourt/interfaces.go

```diff
@@ -42,7 +42,8 @@ type OnionProcessor interface {
 	// ReconstructHopIterator attempts to decode a valid sphinx packet from
 	// the passed io.Reader instance.
 	ReconstructHopIterator(r io.Reader, rHash []byte,
-		blindingKey *btcec.PublicKey) (hop.Iterator, error)
+		blindingKey *btcec.PublicKey, incomingAmt lnwire.MilliSatoshi,
+		incomingExpiry uint32) (hop.Iterator, error)
```

Comment:
A matter of preference, but in this case, all the added arguments are in fact optional, so they could be reflected as such. Then the reader gets that hint at this site vs needing to dig into the actual concrete implementation to realize they're only used if the blinding key itself isn't nil.

Reply:
As with `BlindingKit` above, we always want to pass these in for the case where we're the introduction node and there's a blinding point in our blob - cleaned up in the same way, removed the pointer + used an `Option` for the `BlindingPoint`